src/composeRestyleFunctions.ts

```diff
@@ -46,16 +46,21 @@ const composeRestyleFunctions = <
       dimensions,
     }: {
       theme: Theme;
-      dimensions: Dimensions;
+      dimensions: Dimensions | null;
     },
   ): RNStyle => {
-    const styles = Object.keys(props).reduce(
-      (styleObj, propKey) => ({
-        ...styleObj,
-        ...funcsMap[propKey as keyof TProps](props, {theme, dimensions}),
-      }),
-      {},
-    );
+    const styles: ViewStyle = {};
+    const options = {theme, dimensions};
+    // We know props won't have prototype properties
+    // eslint-disable-next-line guard-for-in
+    for (const key in props) {
```

Comment:
Can this break if objects inherit some default properties?
For e.g, if object prototype includes some default values?

Reply:
Technically, it _might_ be possible, although I don't expect to happen. The extra if should be fine, added it